frontend/src/components/ui/blockEditor/directoryViewer/AgentPrompt.jsx

```diff
@@ -40,24 +42,37 @@ export default function AgentPrompt() {
     e.preventDefault();
     setIsLoading(true);
 
-    const newPrompt = textArea.trim();
-    const response = await callAgent.mutateAsync({
-      userMessage: newPrompt,
-      agentName: agentName,
-      conversationHistory: chatHistory.history,
-      apiKey: openAIApiKey,
-    });
+    try {
+      const newPrompt = textArea.trim();
+      const data = {
+        userMessage: newPrompt,
+        agentName: agentName,
+        conversationHistory: chatHistory.history,
+        apiKey: openAIApiKey,
+      };
 
-    chatHistory.addPrompt(newPrompt, response);
-    await fileBuffer.updateSave(response);
-    if (fileHandle.isComputation) {
-      compile(pipeline.path, blockId);
-    }
+      console.log(data);
+      const response = await callAgent.mutateAsync(data);
+      console.log(response);
 
-    selectedPrompt.unselect();
+      chatHistory.addPrompt(newPrompt, response);
+      await fileBuffer.updateSave(response);
+      if (fileHandle.isComputation) {
+        compile(pipeline.path, blockId);
+      }
 
-    setTextArea("");
-    setIsLoading(false);
+      selectedPrompt.unselect();
+      setTextArea("");
+    } catch (error) {
+      console.log(error);
+      setCompilationError({
+        show: true,
+        title: error.message,
+        caption: error.data?.details,
+      });
+    } finally {
+      setIsLoading(false);
```

Comment:
the catch block doesn't return, do you need the finally block or could just put `setIsLoading` after the try-catch?

Reply:
done